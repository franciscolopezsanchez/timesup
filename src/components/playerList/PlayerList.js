import React from "react"
import "./PlayerList.css"
import PlayerInput from "../playerInput/PlayerInput"
import Player from "../player/Player"

function PlayerList({ teamId, teamName, players, createPlayer, removePlayer }) {
	if (!players) return null
	return (
		<div className="playerList">
			{teamName && <h3>{teamName}</h3>}
			<ul>
				{players.map(player => {
					return (
						<li key={player.name} className="player">
							<Player
								name={player.name}
								removePlayer={removePlayer}
							/>
						</li>
					)
				})}
				<li className="playerInput">
					<PlayerInput createPlayer={createPlayer} teamId={teamId} />
				</li>
			</ul>
		</div>
	)
}

export default PlayerList
