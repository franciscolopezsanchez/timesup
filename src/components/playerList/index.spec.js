import React from "react"
import {render, cleanup} from "@testing-library/react"
import PlayerList from "."

afterEach(cleanup)

describe("PlayerList tests", () => {
  test("PlayerList handles undefined argument", () => {
    const players = undefined
    const {container} = render(<PlayerList players={players} />)
    expect(container.firstChild).toBeFalsy()
  })

  test("PlayerList handles empty list of players as argument", () => {
    const players = []
    const {container, getByText} = render(
      <PlayerList players={players} teamName="nombre" />
    )
    const header = getByText(/nombre/i)
    expect(container).toBeTruthy()
    expect(header).toBeTruthy()
    expect(document.querySelector(".player")).toBeFalsy()
  })

  test("PlayerList handles a list of players as argument", () => {
    const players = [
      {name: "jesus", team: 0},
      {name: "rocio", team: 0},
      {name: "cris", team: 1},
      {name: "curro", team: 1},
    ]
    const {container, getByText} = render(
      <PlayerList players={players} teamName="nombre" />
    )
    const header = getByText(/nombre/i)
    expect(container).toBeTruthy()
    expect(header).toBeTruthy()
    expect(document.querySelector(".player")).toBeTruthy()
  })

  test("PlayerList handles no teamName as argument", () => {
    const players = []
    const {container} = render(<PlayerList players={players} />)
    const header = container.querySelector("h3")
    expect(container).toBeTruthy()
    expect(header).toBeFalsy()
  })

  test("PlayerList handles an empty teamName as argument", () => {
    const players = []
    const {container} = render(<PlayerList players={players} teamName="" />)
    const header = container.querySelector("h3")
    expect(container).toBeTruthy()
    expect(header).toBeFalsy()
  })
})
