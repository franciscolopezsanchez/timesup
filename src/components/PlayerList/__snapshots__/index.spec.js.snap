// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerList tests renders snapshot 1`] = `
<DocumentFragment>
  <div
    class="playerList"
  >
    <ul>
      <li
        class="player"
      >
        <div
          class="row"
        >
          <img
            alt="jesus"
            class="avatar"
            src="https://api.adorable.io/avatars/50/jesus@adorable.png"
          />
          <span
            class="name"
          >
            jesus
          </span>
        </div>
      </li>
      <li
        class="player"
      >
        <div
          class="row"
        >
          <img
            alt="rocio"
            class="avatar"
            src="https://api.adorable.io/avatars/50/rocio@adorable.png"
          />
          <span
            class="name"
          >
            rocio
          </span>
        </div>
      </li>
      <li
        class="player"
      >
        <div
          class="row"
        >
          <img
            alt="cris"
            class="avatar"
            src="https://api.adorable.io/avatars/50/cris@adorable.png"
          />
          <span
            class="name"
          >
            cris
          </span>
        </div>
      </li>
      <li
        class="player"
      >
        <div
          class="row"
        >
          <img
            alt="curro"
            class="avatar"
            src="https://api.adorable.io/avatars/50/curro@adorable.png"
          />
          <span
            class="name"
          >
            curro
          </span>
        </div>
      </li>
      <li
        class="playerInput"
      />
    </ul>
  </div>
</DocumentFragment>
`;
